rundeckapp/src/main/groovy/com/dtolabs/rundeck/app/support/ExecutionQuery.groovy

```diff
@@ -143,4 +143,287 @@ class ExecutionQuery extends ScheduledExecutionQuery implements Validateable{
             }
         }
     }
+
+
+  /**
+   * Creates a criteria closure for this query.
+   * @param customDelegate sets a custom closure delegate. If not defined the default will be used.
+   * @return A closure to run this query within a criteria of Execution
+   */
+  def createCriteria(def customDelegate = null) {
+
+      def query = this
+      def state = query.statusFilter
+      def txtfilters = ScheduledExecutionQuery.TEXT_FILTERS
+      def eqfilters = ScheduledExecutionQuery.EQ_FILTERS
+      def boolfilters = ScheduledExecutionQuery.BOOL_FILTERS
+      def filters = ScheduledExecutionQuery.ALL_FILTERS
+      def excludeTxtFilters = ['excludeJob': 'jobName']
+      def excludeEqFilters = ['excludeJobExact': 'jobName']
+
+      def jobqueryfilters = ['jobListFilter', 'jobIdListFilter', 'excludeJobListFilter', 'excludeJobIdListFilter', 'jobFilter', 'jobExactFilter', 'groupPath', 'groupPathExact', 'descFilter', 'excludeGroupPath', 'excludeGroupPathExact', 'excludeJobFilter', 'excludeJobExactFilter']
+
+      def convertids = { String s ->
+        try {
+          return Long.valueOf(s)
+        } catch (NumberFormatException e) {
+          return s
+        }
+      }
+      def idlist = query.jobIdListFilter?.collect(convertids)
+      def xidlist = query.excludeJobIdListFilter?.collect(convertids)
+
+      def hasJobFilters = jobqueryfilters.any { query[it] }
+      if (hasJobFilters && query.adhoc) {
+        flash.errorCode = "api.error.parameter.error"
+        flash.errorArgs = [message(code: 'api.executions.jobfilter.adhoc.conflict')]
+        return chain(controller: 'api', action: 'renderError')
```

Comment:
does this work from within this method? handling errors/response should probably be moved out of here

Reply:
e.g. just throw exception here and catch it in the controller

Reply:
improved handling of error messages